service/src/test/java/bio/terra/appmanager/api/AdminControllerTest.java

```diff
@@ -77,6 +83,48 @@ void testCreate_204() throws Exception {
         capture_chartVersions.getValue().get(0), chartName, chartVersion, null, null, null);
   }
 
+  @Test
+  void testInvalidChartVersionPost() throws Exception {
+    String chartName = "chart-name-here";
+    String chartVersion = "invalid-chart-version$";
+
+    mockMvc
+        .perform(
+            post("/api/admin/v1/charts/versions")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(
+                    "[{"
+                        + "\"chartName\": \""
+                        + chartName
+                        + "\","
+                        + "\"chartVersion\": \""
+                        + chartVersion
+                        + "\""
+                        + "}]"))
+        .andExpect(status().isBadRequest());
+  }
+
+  @Test
+  void testInvalidChartNamePost() throws Exception {
```

Comment:
```suggestion
  void testCreate_invalidChartName() throws Exception {
```
... same comment here